apollo-router/src/plugins/traffic_shaping/mod.rs

```diff
@@ -367,24 +358,21 @@ impl TrafficShaping {
         subgraph::Request,
         Response = subgraph::Response,
         Error = BoxError,
-        Future = Either<
-            Either<
-                BoxFuture<'static, Result<subgraph::Response, BoxError>>,
-                Either<
-                    BoxFuture<'static, Result<subgraph::Response, BoxError>>,
-                    timeout::future::ResponseFuture<
-                        Oneshot<
-                            Either<
-                                Retry<RetryPolicy, Either<rate::service::RateLimit<S>, S>>,
-                                Either<rate::service::RateLimit<S>, S>,
-                            >,
-                            subgraph::Request,
-                        >,
-                    >,
-                >,
-            >,
-            <S as Service<subgraph::Request>>::Future,
-        >,
+        Future = TrafficShapingSubgraphFuture<S>, /*Either<
+                                                      Either<
+                                                          BoxFuture<'static, Result<subgraph::Response, BoxError>>,
+                                                          timeout::future::ResponseFuture<
+                                                              Oneshot<
+                                                                  Either<
+                                                                      Retry<RetryPolicy, Either<rate::service::RateLimit<S>, S>>,
+                                                                      Either<rate::service::RateLimit<S>, S>,
+                                                                  >,
+                                                                  subgraph::Request,
+                                                              >,
+                                                          >,
+                                                      >,
+                                                      <S as Service<subgraph::Request>>::Future,
+                                                  >*/
```

Comment:
To delete

Reply:
fixed in [b529853](https://github.com/apollographql/router/pull/4170/commits/b5298535ac0b1754e975ae139831f462d467fe5b)